pkg/engine/executor/executor_test.go

```diff
@@ -97,3 +102,181 @@ func TestExecutor_Projection(t *testing.T) {
+				{"message": "status==value level=error", "level": nil, "status": nil, "__error__": "LogfmtParserErr", "__error_details__": "logfmt syntax error at pos 8 : unexpected '='"},
+				{"message": "level=\"unclosed status=500", "level": nil, "status": nil, "__error__": "LogfmtParserErr", "__error_details__": "logfmt syntax error at pos 27 : unterminated quoted value"},
+			},
+		},
+		{
+			name: "extract all keys when none requested",
+			schema: arrow.NewSchema([]arrow.Field{
+				{Name: types.ColumnNameBuiltinMessage, Type: arrow.BinaryTypes.String},
+			}, nil),
+			input: arrowtest.Rows{
+				{"message": "level=info status=200 method=GET"},
+				{"message": "level=warn code=304"},
+				{"message": "level=error status=500 method=POST duration=123ms"},
+			},
+			requestedKeys:  nil, // nil means extract all keys
+			expectedFields: 6,   // 6 columns: message, code, duration, level, method, status
+			expectedOutput: arrowtest.Rows{
+				{"message": "level=info status=200 method=GET", "code": nil, "duration": nil, "level": "info", "method": "GET", "status": "200"},
+				{"message": "level=warn code=304", "code": "304", "duration": nil, "level": "warn", "method": nil, "status": nil},
+				{"message": "level=error status=500 method=POST duration=123ms", "code": nil, "duration": "123ms", "level": "error", "method": "POST", "status": "500"},
+			},
+		},
+		{
+			name: "extract all keys with errors when none requested",
+			schema: arrow.NewSchema([]arrow.Field{
+				{Name: types.ColumnNameBuiltinMessage, Type: arrow.BinaryTypes.String},
+			}, nil),
+			input: arrowtest.Rows{
+				{"message": "level=info status=200 method=GET"},       // Valid line
+				{"message": "level==error code=500"},                  // Double equals error
+				{"message": "msg=\"unclosed duration=100ms code=400"}, // Unclosed quote error
+				{"message": "level=debug method=POST"},                // Valid line
+			},
+			requestedKeys:  nil, // nil means extract all keys
+			expectedFields: 6,   // 6 columns: message, level, method, status, __error__, __error_details__
+			expectedOutput: arrowtest.Rows{
+				{"message": "level=info status=200 method=GET", "level": "info", "method": "GET", "status": "200", "__error__": nil, "__error_details__": nil},
+				{"message": "level==error code=500", "level": nil, "method": nil, "status": nil, "__error__": "LogfmtParserErr", "__error_details__": "logfmt syntax error at pos 7 : unexpected '='"},
+				{"message": "msg=\"unclosed duration=100ms code=400", "level": nil, "method": nil, "status": nil, "__error__": "LogfmtParserErr", "__error_details__": "logfmt syntax error at pos 38 : unterminated quoted value"},
+				{"message": "level=debug method=POST", "level": "debug", "method": "POST", "status": nil, "__error__": nil, "__error_details__": nil},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create input data with message column containing logfmt
+			input := NewArrowtestPipeline(
+				memory.DefaultAllocator,
```

Comment:
nit: You may with to create a checked allocator here to assert that we do releases correctly:

```go
alloc := memory.NewCheckedAllocator(memory.DefaultAllocator) 
defer alloc.AssertSize(t, 0) // Assert empty on test exit 

input := NewArrowtestPipeline(alloc, tt.schema, tt.input) 

// ... 

pipeline := NewParsePipeline(parseNode, input, alloc) 
```